### ISSUE TITLE
Different Arabic fonts not working, only English do.

### ISSUE THREAD

**Issue by TarekAlQaddy** [wkhtmltopdf/wkhtmltopdf#3954]
Importing external Arabic fonts but they didn't work. It's only working on English fonts.

I tried importing fonts from google fonts with `link` tag or `@import` in CSS but no results.

**Comment by PhilterPaper** [comment]
Can you elaborate on "didn't work"? Are you seeing nothing at all, or basic Arabic characters that aren't correctly joined (positioned) or substituted (initial, medial, etc.)? Have you been able to get it to work with locally _installed_ Arabic fonts (not importing from the outside)? Exactly what version of wkHTMLtoPDF are you using, and on what platform? Do the HTML pages work OK on a browser, and exactly how are you linking or importing the fonts? Are you getting any error messages from wkHTMLtoPDF? Can you tell if the fonts are being included in the PDF produced, or is it depending on a local installation of those fonts? I'm not sure if a resulting PDF is supposed to be able to import a _remote_ font, as a browser can (i.e., it relies on a local installation). Maybe someone will be able to help you with sufficient information.

By the way, "English" fonts are more correctly referred to as the _Latin_ alphabet.

**Comment by TarekAlQaddy** [comment]
@PhilterPaper 
I'm seeing the default font for arabic when importing the font.
No I haven't been able to make it work with locally installed fonts.
wkHTMLtoPDF version is: 0.12.3 (with patched qt).
Linux Ubuntu 64x v17.04
I'm running rails and there is no error in console.

For more illustration: 
this is without importing the font
![image](https://user-images.githubusercontent.com/17699683/41506830-876f0ca8-7225-11e8-9154-06d2621ce82d.png)

this is with importing [Cairo font](https://fonts.google.com/specimen/Cairo)

![image](https://user-images.githubusercontent.com/17699683/41506834-92bafa2c-7225-11e8-92ba-b09922cc9c40.png)

**Comment by PhilterPaper** [comment]
Are you saying that the first example ("without importing the font") and the second example ("importing Cairo font") are the same? I can see that the **54#** is different, with the second being in Cairo font. Other numbers appear to be Cairo, too, so that font definitely _is_ being loaded. As for the rest of the text being in the default font rather than Cairo, how are you specifying in CSS the font to be used? Maybe your text doesn't say anything about selecting a font?

**Comment by TarekAlQaddy** [comment]
Yes exactly, the font is being loaded but the effect is only on Latin alphabet or numbers, I'm defining the font globally in `body` css selector. Also note that The text and numbers are beside each other no difference in writing them in HTML, they are in the same HTML tag.

That would behave as expected if you render it on a browser.

**Comment by PhilterPaper** [comment]
The only thing I can think of is that it's an encoding issue. Are the HTML _and_ the font UTF-8, or are you using ISO-8859-x for either? What is the format of the font: TTF or something else? If WebKit is going to the default Arabic font because it can't find the desired Arabic glyph, but a browser _can_ find the glyph, that could be a problem in WebKit. In that case, switching to a different rendering engine (such as Chrome) may be your solution.

**Comment by rome-77** [comment]
@TarekAlQaddy i am facing the same issue. Any updates to solutions on this?

**Comment by TarekAlQaddy** [comment]
@rome-77 No, I made it with the default font for now.

**Comment by singhmss** [comment]
I had imported the Cairo font and found that it was applied on Arabic text in Chrome and Firefox but not in IE11 and Edge. On reviewing the Cairo font details on fonts.google.com, I found a button on the top right side of the page to select the font. This opens a drawer at the bottom of the screen where  options can be selected to generate a customised URL to embed the font on your page. Under Customize > Languages select Arabic. This modifies the standard URL by adding a query string subset=arabic. Using this customised URL the Cairo font was applied on Arabic text in IE11 and Edge.
![Cairo customise 1](https://user-images.githubusercontent.com/58115952/70808368-95b08c80-1dd0-11ea-8507-f370fb5b8906.png)
![Cairo customise 2](https://user-images.githubusercontent.com/58115952/70808381-9ea15e00-1dd0-11ea-94b4-7a0e1f2b3c36.png)
